File: modules/spring-core/testdata/inspection/autowired/src/component_scan/constructor/TestAutowired.java
package component_scan.constructor

import org.springframework.stereotype.Component;
import org.springframework.stereotype.Service;
import org.springframework.beans.factory.annotation.Autowired;

@Service
public class MyComponent {

    @Autowired
    public MyComponent(BarBean bean) {
    }

    @Autowired
    public MyComponent() {
    }

    @Autowired
    public MyComponent(FooBean foobean, BarBean bean) {
    }

    @Autowired(required = false)
    public MyComponent(FooBean foobean, BarBean bean, String str) {
    }
}

public class BarBean {
}

@Component
public class FooBean {
}

@Service
public class MyFactory {
    public MyFactory(String str) {
    }
    public MyFactory(int count) {
    }
}

File: modules/spring-core/testdata/kotlin/inspection/autowired/src/component_scan/constructor/TestAutowired.kt
package component_scan.constructor

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import org.springframework.stereotype.Service

@Service
open class MyComponent {
    @Autowired
    constructor(bean: BarBean?)

    @Autowired
    constructor()

    @Autowired
    constructor(foobean: FooBean?, bean: BarBean?)

    @Autowired(required = false)
    constructor(foobean: FooBean?, bean: BarBean?, str: String?)
}

@Component
open class FooBean

@Service
open class MyFactory {
    constructor(str: String?)
    constructor(count: Int)
}

@Component
class ConstructorProperties(
    var param: String? = null
)